CMSgov/onemac
# OY2 20367: BUG - Spinning circle appears in browser after clicking Submit button for CHIP SPA

Story: https://qmacbis.atlassian.net/browse/OY2-20367
Endpoint: See github-actions bot comment

### Details
User is getting stuck at spinner when attempting to submit a CHIP SPA with several very small attachments.  Initial triage discovered this is the error we have seen before where AWS throttles the uploads and we do not receive the error message.

Apparently Amplify has updated and we now do get the error message returned for the 503!

### Changes
- fixed a bug in the catch that wrongly assumed an error was a string
- updated the error handling to pass back a RESPONSE_CODE so the error gets shown as a red alert (going to be hard to test that, I tested it by forcing a fail condition on the function in the code itself because I ran out of other ideas)
- changed the upload function so it retries an upload 10 times (9?) after receiving the 503 error

### Test Plan
1. Login as a submitting user
2. Navigate to Dashboard
3. Click "New Submission->State Plan Amendment(SPA)->CHIP SPA"
4. use a valid new CHIP SPA id
5. Attach small documents to most of the types (user did all but one, max file size was 1.2MB)
6. Open Inspect
7. Submit the form... if there is an error, verify that the submission was, in fact, successful
8. If no error, you might have to wait a while before retrying as I believe this error may be happening because AWS will send 503 messages to slow down if it decides to add more resources because it takes a moment for those to initialize... there are other reasons, but that one seems most likely to me....
